src/main/scala/org/hammerlab/guacamole/reference/MapBackedReferenceSequence.scala

```diff
@@ -0,0 +1,19 @@
+package org.hammerlab.guacamole.reference
+
+import org.hammerlab.guacamole.Bases
+
+/**
+ * A ContigSequence implementation that uses a Map to store only a subset of bases. This is what you get if you load
+ * a "partial fasta".
+ *
+ * @param wrapped
+ */
+case class MapBackedReferenceSequence(length: Int, wrapped: Map[Int, Byte]) extends ContigSequence {
+  def apply(index: Int): Byte = wrapped.getOrElse(index, Bases.N)
+
+  override def iterator: Iterator[Byte] = {
```

Comment:
I think this may be a confusing method to have implemented since it will iterate only through the bases that are in the map, right? So the user would have no idea what site each base corresponds to. I think either need to start at 0 or just throw an error here